### DISCUSSION TITLE
Clarification on how teams works

### DISCUSSION THREAD

**Discussion by mps-pso** [spatie/laravel-permission#2738] [Q&A] [answered]
I'm a little confused as to how the teams feature actually works and I hope someone can help clarify it for me.

It's my understanding that a user can belong to any of the below scenarios:

1.  A user can be a member of one or more Teams. In turn, a Team has Roles and a Role has Permissions.
2. A user does not need to be a member of a Team and can directly have a Role (a global role).
3. A user can belong to a Team (with a subsequent Role and Permissions) *AND* have a Role (a global role) not associated to a Team.

Am I correct in this understanding?

If point 2 and 3 are correct, why is the `team_id` field not `nullable` in the `model_has_roles` field?

**Comment by parallels999** [comment]
point 2 is incorrect, A user does need to be a member of a Team, 
and Role and Permissions must be assigned on respective team, https://github.com/spatie/laravel-permission/discussions/2731#discussioncomment-10961994

**Reply by mps-pso** [reply]
Got you, thank you for replying. So in essence, if Teams feature is used, each user has to be a member of a Team in order to make use of the Roles/Permissions?

What about point 3? If I try setting a User with a global role, I get the expected:

'SQLSTATE[HY000]: General error: 1364 Field 'team_id' doesn't have a default value'
`insert into `model_has_roles` (`model_id`, `model_type`, `role_id`) values (2, App\Models\User, 6)`

I'd expect that if a User can be have a Role that isn't associated with a team then the `team_id` field in `model_has_roles` table should be `nullable`?

**Reply by parallels999** [reply, marked as answer]
>A user does need to be a member of a Team,
and Role and Permissions must be assigned on respective team

A user always belong to a Team, the Role can be global(_Means that it can be assigned on all teams but not without a team_)
So, you must set the team to assign the global role


https://spatie.be/docs/laravel-permission/v6/basic-usage/teams-permissions#content-rolespermissions-assignment--removal

**Reply by mps-pso** [reply]
Perfect, thank you. I had an epiphany just now in that perhaps 'Global' simply means applicable globally to teams, not outside the scope of teams. Thank you again for your help.
